### PR TITLE
[Testing] Upgrade the SequencerInbox tests to UUPS upgrade mechanism

### PR DESCRIPTION
**Pull request by saxenism**
# Goals of PR

Upgraded the SequencerInbox tests to UUPS upgrade mechanism following the changes in the core contract upgrade mechanisms

### CODE REVIEW

File: contracts/test/SequencerInbox.t.sol [hunk 1]
```diff
@@ -19,24 +19,24 @@
 pragma solidity ^0.8.13;
 
 import "forge-std/Test.sol";
-import "@openzeppelin/contracts/proxy/transparent/TransparentUpgradeableProxy.sol";
 import "../src/ISequencerInbox.sol";
 import "../src/libraries/Errors.sol";
+import {SpecularProxy} from "test/SpecularProxy.sol";
```
Comment (LEAFERx) [inline contracts/test/SequencerInbox.t.sol:24]: just import `"@openzeppelin/contracts/proxy/ERC1967/ERC1967Proxy.sol"`

**Review by LEAFERx — CHANGES_REQUESTED** [review]

File: contracts/test/SequencerInbox.t.sol (hunk 1 above)
Comment (saxenism) [reply contracts/test/SequencerInbox.t.sol]: Done. Please review again.

Comment (ujvl) [reply contracts/test/SequencerInbox.t.sol]: can you delete SpecularProxy now

Comment (saxenism) [reply contracts/test/SequencerInbox.t.sol]: I am not sure what `forge diff` means?

**Review by ujvl — APPROVED** [review]

File: contracts/test/SequencerInbox.t.sol [hunk 2]
```diff
@@ -19,24 +19,24 @@
...
     address internal bob;
+    address internal sequencerOwner;
+    address internal sequencerAddress;
 
     function setUp() public virtual {
-        sequencer = makeAddr("sequencer");
-
         alice = makeAddr("alice");
-
         bob = makeAddr("bob");
```
Comment (LEAFERx) [inline contracts/test/SequencerInbox.t.sol:37]: If these accounts are not used pls delete them

**Review by LEAFERx — APPROVED** [review]

File: contracts/test/SequencerInbox.t.sol (hunk 2 above)
Comment (ujvl) [reply contracts/test/SequencerInbox.t.sol:37]: @saxenism can you do this and then we can merge

Comment (saxenism) [reply contracts/test/SequencerInbox.t.sol]: Done. Please review again.

**Pull request merged**